### PR TITLE
Converts the Distribution CRUD tests to use pytest fixtures

### PR DESCRIPTION
**Pull request by dkliban**
[noissue]

### CODE REVIEW

File: pulp_file/tests/functional/api/from_pulpcore/test_distributions.py [hunk 1]
```diff
@@ -11,240 +13,141 @@
...
+):
+    def create_distribution_and_assert(data):
+        with pytest.raises(ApiException) as exc:
+            file_distro_api_client.create(data)
+        assert json.loads(exc.value.body)["base_path"] is not None
+
+    def update_distribution_and_assert(distribution_href, data):
+        with pytest.raises(ApiException) as exc:
+            file_distro_api_client.update(distribution_href, data)
+        assert json.loads(exc.value.body)["base_path"] is not None
```
Comment (bmbouter) [inline pulp_file/tests/functional/api/from_pulpcore/test_distributions.py:124]: I think moving these to the module level and prepending an underscore would be nice.

File: pulp_file/tests/functional/api/from_pulpcore/test_distributions.py [hunk 2]
```diff
@@ -11,240 +13,141 @@
...
+    assert distribution.content_guard is None
+    assert distribution.repository is None
+    assert distribution.publication == publication.pulp_href
+    assert distribution.base_path == distribution_data["base_path"]
+    assert distribution.name == distribution_data["name"]
+
+    # Assert that the publication has a reference to the distribution
+    assert publication.distributions[0] == distribution.pulp_href
+
+    # Test updating name and base_path with 'partial_update'
```
Comment (bmbouter) [inline pulp_file/tests/functional/api/from_pulpcore/test_distributions.py:72]: Breaking this comment into two comments to cover the two blocks would be nice. Also similarly below.

File: pulp_file/tests/functional/api/from_pulpcore/test_distributions.py (hunk 1 above)
Comment (dkliban) [reply pulp_file/tests/functional/api/from_pulpcore/test_distributions.py:124]: I put them in here so I wouldn't have to pass the client object into them. I don't mind moving to module level, but I did want to share the motivation.

Comment (dkliban) [reply pulp_file/tests/functional/api/from_pulpcore/test_distributions.py]: I updated to using module level methods.

Comment (bmbouter) [reply pulp_file/tests/functional/api/from_pulpcore/test_distributions.py]: Your explanation gives me closure. Get it!? :)

**Review by bmbouter — APPROVED** [review]
This is great. Thank you!

**Pull request merged**